### PR TITLE
Restructure for Weekly Build

### PR DESCRIPTION
**Pull request by forsyth2**
- Update README to specify differences between the weekly build (Weekly Develop) and the nightly build (Nightly Develop), as well as add clarification for running the fork builds (the individual plans).
- Remove accuracy tests since the performance tests include an accuracy test.
- Update default executable to use each plan's own executable instead of Nightly Develop's.
- Add `--weekly` option to specify a test should only be run on the weekly build.
- Update expected performance and accuracy numbers, using lenet and alexnet; remove mnist_distributed_io and resnet50 tests.
- Update the calls to 'salloc' and 'srun' to reduce the amount of indirection (e.g. if the model is alexnet, use these parameters - as opposed to passing in all those parameters).
- Update comments to clarify meaning of 'salloc' and 'srun' parameters.
- Print output file if the model crashes.
- Add optional logging to give more details about actual performance and accuracy numbers.
- Update regex to extract actual performance and accuracy numbers.
- Remove lenet checkpoint callback.

### CODE REVIEW

File: bamboo/integration_tests/performance_tests/test_tools.py [hunk 1]
```diff
@@ -93,52 +89,56 @@ def csv_to_dict(csv_path):
...
-  performance = get_performance(output_file_name)
-  run_tests(performance, model_name, dir_name)
-
-def resnet50_skeleton(executable, dir_name):
-  model_name = 'resnet50'
-  output_file_name = '%s_output.txt' % model_name
-  run_lbann(model_name=model_name, reader_name='imagenet', optimizer_name='adagrad', output_file_name=output_file_name, executable=executable, dir_name=dir_name, num_epochs=1, procs_per_model=2)
-  performance = get_performance(output_file_name)
-  run_tests(performance, model_name, dir_name)
+  run_lbann(model_name=model_name, reader_name='imagenet', optimizer_name='adagrad', output_file_name=output_file_name, executable=executable, dir_name=dir_name, num_nodes=16, timeout=600, num_processes=32, procs_per_model=16)
```
Comment (forsyth2) [inline bamboo/integration_tests/performance_tests/test_tools.py]: @timmoon10 Are these parameters correct? We're running alexnet on 16 nodes, with 32 processes and 16 processes per model (=> 32/16=2 models).

File: bamboo/integration_tests/conftest.py [hunk 2]
```diff
@@ -1,15 +1,21 @@
-import pytest, re, subprocess
+import pytest, os, re, subprocess
 
 def pytest_addoption(parser):
     cluster = re.sub('[0-9]+', '', subprocess.check_output('hostname'.split()).strip())
     default_dirname = subprocess.check_output('git rev-parse --show-toplevel'.split()).strip()
-    default_exe = '%s/../LBANN-NIGHTD-BDE/build/%s.llnl.gov/model_zoo/lbann' % (default_dirname, cluster)
+    # plan = os.environ['bamboo.planKey']
```
Comment (forsyth2) [inline bamboo/integration_tests/conftest.py]: Remove these commented lines.

File: bamboo/integration_tests/performance_tests/expected_performance.csv [hunk 3]
```diff
@@ -1,4 +1,3 @@
-Model_name,           max_epoch_time, max_mean_minibatch_time, max_min_time, max_max_time, max_stdev, min_accuracy
-mnist_distributed_io, 50.00,          0.02,                    0.10,         0.10,         0.05,      96.00
-alexnet,              700.00,         200.00,                  20.00,        50.00,        40.00,     8.00
-resnet50,             2500.00,        75.00,                   40.00,        75.00,        5.00,      7.00
+Model_name,  max_epoch_time, max_mean_minibatch_time, max_max_time, max_min_time, max_stdev, min_accuracy
+alexnet,     254.21,         0.52,                    3.50,         0.43,         0.21,      1.03
```
Comment (forsyth2) [inline bamboo/integration_tests/performance_tests/expected_performance.csv]: Update numbers so the tests pass.
